packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -591,14 +591,14 @@ export abstract class BaseSegment implements ISegment {
 
 			case MergeTreeDeltaType.OBLITERATE:
 				const moveInfo: IMoveInfo | undefined = toMoveInfo(this);
-				assert(moveInfo !== undefined, 0x86e /* On obliterate ack, missing move info! */);
```

Comment:
Looks like this assert got removed entirely accidentally and should be restored.